src/listfunc.c

```diff
@@ -160,7 +160,8 @@ Obj FuncADD_LIST3 (
   } else if ( TNUM_OBJ( list ) < FIRST_EXTERNAL_TNUM ) {
     AddList3( list, obj, ipos );
 #ifdef HPCGAP
-  } else if ( TNUM_OBJ(list) == T_ALIST ) {
+  // Only support adding to end of atomic lists
+  } else if ( TNUM_OBJ(list) == T_ALIST && pos == (Obj)0 ) {
     AddAList( list, obj );
```

Comment:
This fix is of course correct, but do we really want to fall-through to regular method dispatch? The alternative would be to move the check for `pos == 0` inside this `else if`, and if `pos != 0`, then we directly call `ErrorMayQuit` with a message like "Inserting elements at a specific position in an atomic list is not possible" or so.

This would users who run into the problem but don't know what's going on -- they still might not know why this operation is not supported, but now at least they can ask a better question.

Reply:
While I agree in principle, there is a general pattern here, there are a lots of list methods which fixed and variable sized atomic lists don't support, and they all end up in regular method dispatch.